site/source/docs/porting/Audio.rst

```diff
@@ -0,0 +1,143 @@
+
+You can reasonably expect ported OpenAL applications to "just work" with no additional effort.  
+However, there are some implementation-specific aspects which are worth considering - these are documented here.
+
+
+.. _Audio-openal-extensions-g:
+
+Supported OpenAL extensions
+===========================
+
+The following extensions are supported by Emscripten's OpenAL implementation.
+
+- ``ALC_SOFT_pause_device``;
+- ``ALC_SOFT_HRTF``;
+- ``AL_EXT_float32``;
+- ``AL_SOFT_loop_points``;
+- ``AL_SOFT_source_length``;
+- ``AL_EXT_source_distance_model``;
+- ``AL_SOFT_source_spatialize``;
+
+.. warning:: This doesn't mean you should just assume their presence ! For correctness, you should **always** check if an extension is supported before using it, like a good application would do.
+
+
+.. _Audio-guidelines-g:
+
+Guidelines for Audio on Emscripten
+==================================
+
+First: Do not loop on OpenAL calls.
+
+Put simply, avoid completely this kind of code :
+
+.. code-block:: c
+
+    while(nframes < THE_NUMBER_OF_FRAMES_WE_WANT)
+        alcGetIntegerv(device, ALC_CAPTURE_SAMPLES, 1, &nframes);
+
+
+The reason for this is simple: Your application needs to yield to the Javascript main loop for audio processing to take place (See :ref:`Browser main loop <emscripten-runtime-environment-main-loop>`).
+
+The above snippet usually works because most OpenAL implementations own and manage one or more separate threads. This is not the case in Emscripten.
+
+What you must do instead is perform each such query only once per "main loop iteration" (i.e the callback you provide via :c:func:`emscripten_set_main_loop` or :c:func:`emscripten_set_main_loop_arg`).
+
+
+Second: Avoid creating and destroying resources relentlessly.
+
+On some browsers (except, apparently, Firefox), garbage collection does not occur as much as it should. Knowing this, your app should ideally at most use one ``ALCcontext`` for playback and one ``ALCdevice`` for capture.
+
+Even if you're planning to suspend audio capture or playback for extended periods of time, it is probably better to keep contexts in memory (unlike in native apps where you should rather destroy them) to avoid risking memory leaks caused by poorly garbage-collected Javascript implementations.
+
+Of course, the current OpenAL implementation strives to be memory-efficient, but it has no control over lower-level mechanisms involved.
+
+
+.. _Audio-openal-capture-behavior-g:
+
+Emscripten-specific capture behavior
+====================================
+
+Attempting to open an input stream to the user's audio capture device
+results in the asynchronous appearance of a small browser-specific dialog
+asking for the user's permission.
+
+.. note:: 
+    On Firefox, the user is also given the opportunity to pick from a list of devices at that moment.
+    On other browsers, this is normally a simple "Allow/Deny" button pair on a toast.
```

Comment:
This browser specific note is a candidate to get out of date on the long run, and does not seem to carry particularly important value, so would be cleaner to avoid this? (or perhaps remark above `asking for the user's permission, and on some browsers, the capture device to choose)`.